packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -28,21 +28,46 @@ export class State {
    */
   public readonly stateName: string;
 
+  private transitionEvents: ITransitionEvent[] = []
+
   constructor(private readonly props: StateProps) {
     this.stateName = props.stateName;
   }
 
   /**
-   * Return the state property JSON
+   * Add a transition event to the state.
+   *
+   * @param transitionEvent the transition event that triggered if condition is evaluated to TRUE
+   */
+  public transitionTo(transitionEvent:ITransitionEvent) {
+    this.transitionEvents.push(transitionEvent);
+  }
+
+  /**
+   * Return the JSON of the states that is transited to.
+   * This function is called recursively and collect the states.
    *
    * @internal
    */
-  public _toStateJson(): CfnDetectorModel.StateProperty {
```

Comment:
Instead of this renaming, leave `_toStateJson()` as is, and introduce a new private method called `_getStatesJson()`, that `_toStateJson()` calls with an empty Set (see above).